drivers/hmis/app/jobs/hmis/auto_exit_job.rb

```diff
@@ -0,0 +1,84 @@
+###
+# Copyright 2016 - 2023 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+###
+
+module Hmis
+  class AutoExitJob < BaseJob
+    def perform
+      Hmis::Hud::Project.hmis.each do |project|
+        config = Hmis::AutoExitConfig.config_for_project(project)
+        next unless config.present?
+        raise "Auto-exit config unusually low: #{config.length_of_absence_days}" if config.length_of_absence_days < 30
+
+        project.enrollments.open_excluding_wip.each do |enrollment|
+          most_recent_contact = if project.project_type == 1 # Night-by-night Emergency Shelter
+            enrollment.services.bed_nights.order(:date_provided).last
+          else
+            [
+              enrollment.services.order(:date_provided).last,
+              enrollment.custom_services.order(:date_provided).last,
+              enrollment.current_living_situations.order(:information_date).last,
+              enrollment.custom_assessments.order(:assessment_date).last,
+              enrollment,
+            ].compact.max_by { |entity| contact_date_for_entity(entity) }
+          end
+
+          most_recent_contact_date = contact_date_for_entity(most_recent_contact)
+          next unless (Date.current - most_recent_contact_date).to_i >= config.length_of_absence_days
+
+          auto_exit(enrollment, most_recent_contact)
+        end
+      end
+    end
+
+    private
+
+    def auto_exit(enrollment, most_recent_contact)
+      exit_date = contact_date_for_entity(most_recent_contact)
+      # If most recent contact was a Bed Night service, the Exit Date should be the day after they received service
+      exit_date += 1.day if most_recent_contact.is_a?(Hmis::Hud::Service) && most_recent_contact.record_type == 200
```

Comment:
so this would argue for returning the full record (from above) if we need to know which kind of record it was.

Reply:
I suppose the extra day could be added when fetched, since we know the source at that time.